### PR TITLE
use recursion to reverse a Linked List

### PR DESCRIPTION
**Pull request by Hiba-alkurd**
Reverse a Linked List Problem

### CODE REVIEW

File: Reverse Linked List/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿Problem 1: Reverse Linked List
+Description:
+Reverse a singly linked list.
+
```
Comment (ziadadeela) [inline Reverse Linked List/README.md:4]: Let's try to add some info about how did we solve it.

File: Reverse Linked List/Main.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿
+//this code works for leetcode submission
+//for the runnable code on IDE see Problems/ReverseLinkedList.cs
+class Program
+{
+    public ListNode Head = null;
```
Comment (ziadadeela) [inline Reverse Linked List/Main.cs]: Is this needed?

File: Reverse Linked List/Main.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        return new_head;
+    }
+}
+
+//[1,2,3,4,5]
+
+public class Solution {
+    public ListNode ReverseList(ListNode head) {
+        if(head == null) return null;
+        Program List = new Program();
```
Comment (ziadadeela) [inline Reverse Linked List/Main.cs]: Why is this needed? :/

File: Reverse Linked List/Main.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        return List.ReverseList(null, head);
```
Comment (ziadadeela) [inline Reverse Linked List/Main.cs]: Can we get rid of this method definition and solve it recursively using:
 ListNode ReverseList(ListNode head)?

File: Problems/Problems/ReverseLinkedList.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                return;
+            }
+            ReverseList(head, head.next);
+            head.next = prev;
+        }
+        
+
+        static void Main(string[] args)
+        {
+            Program List = new Program();
```
Comment (ziadadeela) [inline Problems/Problems/ReverseLinkedList.cs]: Naming convention: list [camelCase]
Check this out:
https://github.com/ktaranov/naming-convention/blob/master/C%23%20Coding%20Standards%20and%20Naming%20Conventions.md

File: Problems/Problems/ReverseLinkedList.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Problems
+{
+    
+    // Definition for singly-linked list.
+
+    public class Node {
```
Comment (ziadadeela) [inline Problems/Problems/ReverseLinkedList.cs]: Let's have a separate class for LinkedList data structure and methods.
Node, AddNode, PrintList, ReverseList

File: Reverse Linked List/Main.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
+﻿
+//this code works for leetcode submission
+//for the runnable code on IDE see Problems/ReverseLinkedList.cs
+class Program
+{
+    public ListNode Head = null;
+
+    //recursive
+    public ListNode ReverseList(ListNode prev, ListNode head)
```
Comment (ziadadeela) [inline Reverse Linked List/Main.cs]: Seems like this method is duplicated?

File: Problems/Problems/ReverseLinkedList.cs [hunk 8]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
```
Comment (ziadadeela) [inline Problems/Problems/ReverseLinkedList.cs]: Why do we have Problems/Problems subfolders?

**Review by ziadadeela — CHANGES_REQUESTED** [review]

File: Reverse Linked List/Main.cs (hunk 3 above)
Comment (Hiba-alkurd) [reply Reverse Linked List/Main.cs]: at first, I thought I could write the solution method signature however I liked, as in HackerRank
i was thrown off a bit by LeetCode because writing and printing wasn't done by me.
so it looks like a patched code

File: Problems/Problems/ReverseLinkedList.cs (hunk 8 above)
Comment (Hiba-alkurd) [reply Problems/Problems/ReverseLinkedList.cs]: this is a testing project, initially, I wanted to keep it untracked by Git then left it in case I needed it

Comment (Hiba-alkurd) [reply Problems/Problems/ReverseLinkedList.cs]: but all the needed files are in a folder with the problems name

**Review by ziadadeela — APPROVED** [review]
